assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js

```diff
@@ -77,21 +77,75 @@ function DashboardTopEarningPagesWidget( { Widget, WidgetReportZero, WidgetRepor
 	} );
 
 	if ( loading ) {
-		return <PreviewTable rows={ 5 } padding />;
+		return (
+			<Widget
+				noPadding
+				Footer={ () => (
+					<SourceLink
+						className="googlesitekit-data-block__source"
+						name={ _x( 'Analytics', 'Service name', 'google-site-kit' ) }
+						href={ analyticsMainURL }
+						external
+					/>
+				) }
+			>
+				<PreviewTable rows={ 5 } padding />
+			</Widget>
+		);
 	}
 
 	// A restricted metrics error will cause this value to change in the resolver
 	// so this check should happen before an error, which is only relevant if they are linked.
 	if ( ! isAdSenseLinked ) {
-		return <AdSenseLinkCTA />;
+		return (
+			<Widget
+				noPadding
+				Footer={ () => (
+					<SourceLink
+						className="googlesitekit-data-block__source"
+						name={ _x( 'Analytics', 'Service name', 'google-site-kit' ) }
+						href={ analyticsMainURL }
+						external
+					/>
+				) }
+			>
+				<AdSenseLinkCTA />
+			</Widget>
+		);
 	}
 
 	if ( error ) {
-		return <WidgetReportError moduleSlug="analytics" error={ error } />;
+		return (
+			<Widget
+				noPadding
+				Footer={ () => (
+					<SourceLink
+						className="googlesitekit-data-block__source"
+						name={ _x( 'Analytics', 'Service name', 'google-site-kit' ) }
+						href={ analyticsMainURL }
+						external
+					/>
+				) }
+			>
+				<WidgetReportError moduleSlug="analytics" error={ error } />
+			</Widget>
+		);
 	}
 
 	if ( isZeroReport( data ) ) {
-		return <WidgetReportZero moduleSlug="analytics" />;
+		<Widget
+			noPadding
+			Footer={ () => (
+				<SourceLink
+					className="googlesitekit-data-block__source"
+					name={ _x( 'Analytics', 'Service name', 'google-site-kit' ) }
+					href={ analyticsMainURL }
+					external
+				/>
+			) }
+		>
+			return <WidgetReportZero moduleSlug="analytics" />;
+		</Widget>;
```

Comment:
I am not sure that this is going to work :stuck_out_tongue_closed_eyes: Did you test it on your local?

Reply:
Good catch, I guess I didn't save that one properly before committing 🙈

Reply:
Ah, I don't have an AdSense account, so have to move things around to get this code to run. That's why it slipped through the net.

Reply:
@johnPhillips just use the storybook to see those different states :smile: